performance-best-practice.md

```diff
@@ -0,0 +1,510 @@
+---
+title: Configure TiDB for Optimal Performance
+summary: Learn how to optimize TiDB performance by configuring key settings and addressing edge cases.
+---
+
+## Configure TiDB for Optimal Performance
+
+This guide provides essential information on optimizing TiDB for maximum performance. We'll explore:
+- Performance best practices for common workloads
+- Strategies for addressing challenging performance scenarios
+
+> **Note:**
+>
+> These optimization techniques are particularly valuable when pursuing extreme performance in TiDB. However, it's important to note that performance tuning often involves trade-offs between multiple factors, and there is no single "silver bullet" solution. Some of the techniques described here may use experimental features, which are specifically called out. While these optimizations can yield significant performance gains, they may not be suitable for stable production environments and should be used with caution.
+
+## Why This Matters
+
+Optimizing TiDB for peak performance requires careful tuning of various settings. In many cases, achieving optimal performance involves adjusting configurations beyond their default values.
+
+While default settings are chosen for stability, maximizing performance often requires more aggressive configurations and sometimes the use of experimental features. This approach is based on insights gained from numerous real-world deployments and performance optimization efforts.
+
+This guide explains these non-default settings, detailing their benefits and potential trade-offs. It's designed to help you make informed decisions when fine-tuning TiDB for your specific workload requirements.
+
+## Key Settings for Common Workloads
+
+The following suggested settings cover the most common optimizations for improving TiDB performance:
+
+- Enhancing query plan caching
+- Optimizing the query optimizer's behavior
+- More aggressive use of TiKV's Titan storage engine
+
+These settings can significantly boost performance for many common workloads, but as with any optimization, it's essential to test thoroughly in your specific environment.
+
+### System variables
+
+```SQL 
+set global tidb_enable_instance_plan_cache=on;
+set global tidb_instance_plan_cache_max_size=2GiB;
+set global tidb_enable_non_prepared_plan_cache=on;
+set global tidb_ignore_prepared_cache_close_stmt=on;
+set global tidb_enable_inl_join_inner_multi_pattern=on;
+set global tidb_opt_derive_topn=on;
+set global tidb_runtime_filter_mode=LOCAL;
+set global tidb_opt_enable_mpp_shared_cte_execution=on;
+set global tidb_rc_read_check_ts=on;
+set global tidb_guarantee_linearizability=off;
+set global pd_enable_follower_handle_region=on;
+set global tidb_opt_fix_control = '44262:ON,44389:ON,44823:10000,44830:ON,44855:ON,52869:ON';
+```
+
+#### Justifications
+
+| variables| Pro | Cons | 
```

Comment:
```suggestion
| System variable | Description | Note |
```
In technical documentation, it’s generally better to avoid informal or overly casual terms like "pro" and "cons".